src/bokeh/events.py

```diff
@@ -275,6 +277,22 @@ def __init__(self, model: Model | None) -> None:
     def event_values(self) -> dict[str, Any]:
         return dict(**super().event_values(), model=self.model)
 
+class AxisClick(ModelEvent):
+    ''' Announce a location where an axis was clicked.
+
+    '''
+    event_name = 'axis_click'
+
+    value: float | FactorType | None
+
+    def __init__(self, model: Axis | None, value: float | FactorType | None = None) -> None:
```

Comment:
default value here seems like a kludge but lots of other events do this as well, maybe a separate issue to clean this and events tests up some.

Reply:
I wouldn't introduce a wrong API just because other cases are like that. `None` doesn't make sense here, neither as as type nor a default. The same applies to `model`.

Reply:
Well, the issue is that there is some serialization codepath somewhere that seems to assume everything is default constructible. Though I suppose it might even just be a test concession. I will have to try to run things down in more detail.

Reply:
I don't want to work on this in this PR, I would rather have a dedicated broader cleanup in its own PR. I made https://github.com/bokeh/bokeh/issues/14061